lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -162,7 +172,7 @@ class GoogleCloudOutput < BufferedOutput
     attr_reader :running_on_managed_vm
     attr_reader :gae_backend_name
     attr_reader :gae_backend_version
-    attr_reader :service_name
+    attr_reader :resource_type
```

Comment:
`Google::Protobuf::Map` does not support `merge!`. Looking at the implementation, `merge_into_self` behaves like `merge!`, but is restricted to be used internal only:

```
// Used only internally -- shared by #merge and #initialize.
VALUE Map_merge_into_self(VALUE _self, VALUE hashmap) {
```

One option is to store the non-grpc version of `MonitoredResource` with `type` and `labels` along the way, and convert it to the grpc version in the end. Would we prefer that?